### PR TITLE
fix Aggregation Function not found error after #16123

### PR DESCRIPTION
**Pull request by ucasfl**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Continue of [#16123](https://github.com/ClickHouse/ClickHouse/pull/16123)

Detailed description / Documentation draft:

...

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Interpreters/TreeRewriter.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ struct CustomizeAggregateFunctionsSuffixData
         {
             auto properties = instance.tryGetProperties(func.name);
             if (properties && !properties->returns_default_when_only_null)
-                func.name = func.name + customized_func_suffix;
+                func.name = Poco::toLower(func.name) + customized_func_suffix;
```
Comment (alexey-milovidov) [inline src/Interpreters/TreeRewriter.cpp:128]: This is strange, because we have non-lowercase names, e.g. sumWithOverflow.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by ucasFL** [comment]
The problem `MAXOrNull` doesn't work is because that combinators doesn't apply for case insensitive names currently, only for native names, can we cancel this constraint or we shoule keep it? @alexey-milovidov

**Comment by alexey-milovidov** [comment]
@ucasFL Yes, let's cancel this constraint.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**